src/ws.nim

```diff
@@ -535,7 +529,126 @@ func newHttpHeaders*(keyValuePairs:
       else:
         result.table[key] = @[pair.val]
 
-proc newAsyncWebsocketClient*(uri: Uri, protocols: seq[string] = @[]): Future[WebSocket] {.async.} =
+proc generateHeaders(requestUrl: Uri, httpMethod: string,
+    headers: HttpHeaders): string =
+  # GET
+  let upperMethod = httpMethod.toUpperAscii()
+  result = upperMethod
```

Comment:
We should avoid using `result` completely. I know that it's use proliferates through the standard library and other popular nim projects, but as per our internal convention, we've determined that its use is dangerous and leads to all sorts of hard to track bugs. The easiest example to convey this is in the presence of exceptions. If an exception is thrown in the middle of a multi step operation like the one in this proc, `result` ends up in an inconsistent state, one can imagine all sorts of scenarios where the exception is ignored and the incomplete/inconsistent `result` continues being used. The better approach is to simply stick to explicit result variables and `return` statments.

Reply:
Still using `result` here.